RecoTracker/MeasurementDet/plugins/TkPixelMeasurementDet.cc

```diff
@@ -127,17 +127,21 @@ TkPixelMeasurementDet::RecHitContainer TkPixelMeasurementDet::compHits(const Tra
       return result;
     }
 
-    if (ci->maxPixelRow() < xminus)
+    if (ci->localPosition().x() < xminus)
       continue;
     // also check compatibility in y... (does not add much)
-    if (ci->minPixelCol() > yplus)
+    if (ci->localPosition().y() > yplus)
       continue;
-    if (ci->maxPixelCol() < yminus)
+    if (ci->localPosition().y() < yminus)
       continue;
 
     if (data.pixelClustersToSkip().empty() or (not data.pixelClustersToSkip()[index])) {
-      SiPixelClusterRef cluster = detSet.makeRefTo(data.pixelData().handle(), ci);
-      result.push_back(buildRecHit(cluster, ts.localParameters()));
+      if (ci->canImproveWithTrack()) {
```

Comment:
and here we decide to improve or keep it as is (I remind that similar code is already in production for the fitter since ever)